README.md

```diff
@@ -136,6 +137,78 @@ $server->on('error', function (Exception $e) {
 Note that the request object can also emit an error.
 Check out [request](#request) for more details.
 
+### Middleware
+
+Middleware can ba added to the server using [`MiddlewareRunner`](src/MiddlewareRunner.php)
+instead of the the `callable`. A middleware is expected adjure to the following rules:
+
+* It is a `callable`.
+* It accepts `ServerRequestInterface` as first argument and `callable` as second argument.
+* It either returns a promise resolving to a `ResponseInterface`; or it calls the second argument
```

Comment:
It is actually allowed to return a `ResponseInterface` directly (not only a Promise resolving to a `PromiseInterface`).

Reply:
How about this?

> * It is a `callable`.
> * It accepts `ServerRequestInterface` as first argument and optional `callable` as second argument.
> * It returns a `ResponseInterface` (or a `PromiseInterface` resolving to a `ResponseInterface`)
> * It calls `$next($request)` to continue processing the next middleware function or returns explicitly to abort the chain

Reply:
> * It returns a `ResponseInterface` (or a `PromiseInterface` resolving to a `ResponseInterface`)

I'd still write just *promise* because you can return any [thenable which can be consumed](https://github.com/reactphp/promise/blob/master/src/functions.php#L11) by react/promise.

Reply:
Valid point, just link to the promise repo and maybe provide a small example below? :+1:

Reply:
Suggestion: 
> It returns a `ResponseInterface` (or any promise which can be consumed by [Promise\resolve()](https://github.com/reactphp/promise#resolve) resolving to a `ResponseInterface`)

Reply:
No objection, just wanted to point out that if probably makes sense to keep this in line with the first paragraph here: https://github.com/reactphp/http#response

Considering the other chapters, doesn't it make more sense to have this chapter *below* the response chapter? Middleware behave transparently to the `Server`, so most of what is described in "Response" still applies to middleware and the `Server` alike.

Reply:
Makes sense 👍